app/main/views/login.py

```diff
@@ -146,9 +180,178 @@ def update_password(token):
                                **template_data), 400
 
 
+@main.route('/create-user/<string:encoded_token>', methods=["GET"])
+def create_user(encoded_token):
+    form = CreateUserForm()
+    template_data = main.config['BASE_TEMPLATE_DATA']
+
+    token = decode_invitation_token(encoded_token)
+
+    if token is None:
+        flash('token_invalid', 'error')
+        return render_template(
+            "auth/create-user.html",
+            form=form,
+            token=None,
+            email_address=None,
+            supplier=None,
+            **template_data), 400
+    else:
+        user = data_api_client.get_user(email_address=token.get("email_address"))
+
+        if user:
+            return render_template(
+                "auth/update-user.html",
+                user=user["users"],
+                email_address=token['email_address'],
+                supplier_name=token['supplier_name'],
+                token=encoded_token,
+                **template_data), 200
+        else:
+            return render_template(
+                "auth/create-user.html",
+                form=form,
+                email_address=token['email_address'],
+                supplier_name=token['supplier_name'],
+                token=encoded_token,
+                **template_data), 200
+
+
+@main.route('/update-user/<string:encoded_token>', methods=["POST"])
+def submit_update_user(encoded_token):
```

Comment:
With regards to login - you need to have a valid token to be able to see this page. You've been invited and the token is emailed to you. Once thats happened a click through here will log you in. 

A check in here to ensure valid user status (active/not a supplier/etch) should be OK.